cfme/tests/automate/custom_button/test_infra_objects_ansible.py

```diff
@@ -0,0 +1,163 @@
+        "VM_INSTANCE" not in button_group and inventory == "Target Machine"
+    )
+    or (appliance.version < "5.11" and inventory == "Localhost")
+)
+def test_custom_button_ansible_automate_infra_obj(
+    request, appliance, inventory, setup_obj, button_group, ansible_catalog_item_create_empty_file,
+    target_machine, target_machine_ansible_creds,
+):
+    """ Test ansible custom button for with specific inventory execution
+
+    Polarion:
+        assignee: ndhandre
+        initialEstimate: 1/4h
+        startsin: 5.9
+        casecomponent: CustomButton
+        tags: custom_button
+        setup:
+            1. Setup Target Machine with pingable hostname
+            2. Create catalog with ansible catalog item
+        testSteps:
+            1. Create custom button group with the Object type
+            2. Create a custom button with specific inventory
+               (localhost/ Target Machine/ Specific Host)
+            3. Navigate to object Details page
+            4. Check for button group and button
+            5. Select/execute button from group dropdown for selected entities
+            6. Fill dialog with proper credentials and hostname
+            7. Check for the proper flash message
+            8. Check operation perform on target machine or not (here create test file).
+    """
+    group, obj_type = button_group
+
+    if inventory == "Localhost":
+        cred_name = "CFME Default Credential"
+        hostname = appliance.hostname
+        username = credentials["ssh"]["username"]
+        password = credentials["ssh"]["password"]
+    else:
+        cred_name = target_machine_ansible_creds.name
+        hostname = target_machine.hostname
+        username = target_machine.username
+        password = target_machine.password
+
+    # Create button as per inventory
+    button = group.buttons.create(
+        type="Ansible Playbook",
+        playbook_cat_item=ansible_catalog_item_create_empty_file.name,
+        inventory=inventory,
+        hosts=target_machine.hostname if inventory == "Specific Hosts" else None,
+        text=fauxfactory.gen_alphanumeric(start="btn_"),
+        hover=fauxfactory.gen_alphanumeric(start="hover_"),
+    )
+    request.addfinalizer(button.delete_if_exists)
+
+    # For target machine inventory target entity object is created target VM
+    entity = target_machine.vm if inventory == "Target Machine" else setup_obj
+
+    # Navigate to entity object and execute button
+    view = navigate_to(entity, "Details")
+    custom_button_group = Dropdown(view, group.hover)
+    assert custom_button_group.has_item(button.text)
+    custom_button_group.item_select(button.text)
+
+    dialog_view = view.browser.create_view(CredsHostsDialogView, wait="20s")
+    dialog_view.fill({"machine_credential": cred_name})
+
+    with SSHClient(hostname=hostname, username=username, password=password) as client:
+        # Clean file if already available on host
+        client.remove_file(ANSIBLE_FILE)
```

Comment:
Consider adding some assertion around the return here, so that in failure its clear whether the file was removed cleanly or not.